### PR TITLE
Fix cluster level dense vector stats

### PR DESCRIPTION
**Pull request by jimczi**
The cluster level dense vector stats returns the total number of dense vector indices globally including the replicas. This commit fixes the total to only include the value count of the primary indices. This change aligns with the docs stats which also reports the number of primary documents when used in cluster stats. The indices stats API still reports granular results for replicas and primaries so the information is not lost.

### CODE REVIEW

File: docs/reference/rest-api/common-parms.asciidoc [hunk 1]
```diff
@@ -556,6 +556,12 @@ Size of the index in <<byte-units, byte units>>.
 
 `translog`::
 <<index-modules-translog,Translog>> statistics.
+
+`dense_vector`::
+Total number of vectors indexed.
```
Comment (kderusso) [inline docs/reference/rest-api/common-parms.asciidoc:561]:
```suggestion
Total number of dense vectors indexed.
```

File: docs/reference/rest-api/common-parms.asciidoc [hunk 2]
```diff
@@ -556,6 +556,12 @@ Size of the index in <<byte-units, byte units>>.
...
+
+<<indices-refresh,Index refreshes>> can affect this statistic.
```
Comment (kderusso) [inline docs/reference/rest-api/common-parms.asciidoc:563]: 😅

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.stats/10_basic.yml [hunk 3]
```diff
@@ -231,15 +231,13 @@
 
 ---
 "Dense vector stats":
-  - requires:
-      cluster_features: ["gte_v8.10.0"]
-      reason:  "dense vector stats added in 8.10"
+  - skip:
```
Comment (kderusso) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.stats/10_basic.yml:234]: Why did this change from the new requires/cluster_features syntax?

File: docs/reference/rest-api/common-parms.asciidoc [hunk 4]
```diff
@@ -562,6 +562,11 @@ Size of the index in <<byte-units, byte units>>.
 `translog`::
 <<index-modules-translog,Translog>> statistics.
 
+`dense_vector`::
```
Comment (kderusso) [inline docs/reference/rest-api/common-parms.asciidoc:565]: Nitpick: Could we make sure this is in alphabetical order?

**Review by kderusso — APPROVED** [review]
LGTM with one comment

File: docs/reference/rest-api/common-parms.asciidoc (hunk 4 above)
Comment (jimczi) [reply docs/reference/rest-api/common-parms.asciidoc]: https://github.com/elastic/elasticsearch/pull/107962/commits/c4e6978bf20d448cb8f9942609164e2887bc0e23

**Pull request merged**